cloud/scope/vpc_cluster.go

```diff
@@ -221,6 +222,89 @@ func (s *VPCClusterScope) CheckTagExists(tagName string) (bool, error) {
 	return exists != nil, nil
 }
 
+// GetAPIServerPort will return the API Server's port.
+func (s *VPCClusterScope) GetAPIServerPort() int32 {
+	// TODO(cjschaef): Add logic to handle cases not default.
+	return infrav1beta2.DefaultAPIServerPort
+}
+
+// GetControlPlaneSubnetIDs returns all of the Control Plane subnet Id's.
+func (s *VPCClusterScope) GetControlPlaneSubnetIDs() ([]string, error) {
+	subnets := make([]string, 0)
+	// Retrieve the subnet Id's from Status.
+	if s.NetworkStatus() != nil && s.NetworkStatus().ControlPlaneSubnets != nil {
+		for _, subnet := range s.NetworkStatus().ControlPlaneSubnets {
+			subnets = append(subnets, subnet.ID)
+		}
+		// NOTE(cjschaef): We assume all Subnets are in Status at this point, we could perhaps reconcile Status with any defined in Spec (preventing duplicates) to be safe.
+		return subnets, nil
+	}
+
+	// NOTE(cjschaef): If Status was not set or ControlPlaneSubnets was empty, the Control Plane subnet ID's could be retrieved from Spec. However, for now consider this an error, since Subnet reconciliation should have run prior and no tracked Control Plane subnets would be a major issue.
+	return subnets, fmt.Errorf("error no control plane subnets available in status")
+}
+
+// GetLoadBalancerHostName will return the hostname of the cluster's public Load Balancer, assuming only one public Load Balancer was provided. Or, the hostname of the single private Load Balancer (assuming the cluster has no public access and only one private Load Balancer was provided).
+// NOTE(cjschaef): A webhook validation check could help ensure this.
+func (s *VPCClusterScope) GetLoadBalancerHostName() (*string, error) {
+	// If no Status or Load Balancer Status is populated, assume the Load Balancer's are not ready (have not been reconciled), so no hostname will be available.
+	if s.NetworkStatus() == nil || s.NetworkStatus().LoadBalancers == nil || len(s.NetworkStatus().LoadBalancers) == 0 {
```

Comment:
we are returning nil here for `len(s.NetworkStatus().LoadBalancers)` and same codition returns error next step, May be we can simplify or remove if not required.

Reply:
So, the first set of checks looks at Status. Assuming if there is no Status or no LB status, the situation needs to wait for Status to exist or be updated.

The second check is against Spec. If Status was set, but Spec was not, that is an error. Checks are made immediately in the function to short circuit.

Reply:
apologies, Its correct, I just overlooked spec and status.